### DISCUSSION TITLE
Regarding issue in re-sizing of images

### DISCUSSION THREAD

**Discussion by Vigneshwar-10** [ImageMagick/ImageMagick#2066] [Help]
Hi Team

When we tried re-sizing of images where we had 13 threads getting executed at the same timestamp(even same milliseconds), 7 images got re-sized successfully while the remaining 6 failed with exit code = 137. Upon re-triggering those 6 images were processed successfully. Kindly suggest us what could be the issue.

Thanks

**Comment by snibgo** [comment]
IM won't multi-thread across images. So I suppose that multi-threading is done by you, with some tool external to IM.

We probably can't help with other tools, but if you say what you do, perhaps we can make suggestions.

**Comment by Vigneshwar-10** [comment]
Hi Team

As suggested I am hereby sharing the way we use image magick. 
We use image magick for re-sizing of images. We use to call convert command from Java code using threads and execute the command in server. As described earlier, during execution we had passed 13 threads at exact same time-stamp of which 7 got processed and 6 got rejected with exit code as 137. So kindly help us by suggesting what could be the issue.

**Comment by snibgo** [comment]
I don't know what exit code 137 means.

I suspect you are overloading your CPU. By default, each instance of IM is mutithreaded. But you are multithreading IM commands in parallel. Unless your CPU has hundreds of cores, this seems like a bad idea.

When multithreading instances of IM, I suggest you tell IM to be single threaded. You might do this at each command with "-limit thread 1" or as a line in policy.xml.

**Comment by hellomsg** [comment]
I have the parallel issue too. Got the error

```text
[INFO] Starting HTTP server localhost on :::8181
Assertion failed: (splay_tree != (SplayTreeInfo *) NULL), function ResetSplayTreeIterator, file MagickCore/splay-tree.c, line 1474.
```
